the-collab-lab/tcl-72-smart-shopping-list
# 13. As a user, I want to view a list of my shopping list items in order of how soon I am likely to need to buy each of them again so that it’s clear what I need to buy soon

## Description


This PR sorts items in each shopping list based on their purchase urgency. It also adds in the UI a sense of urgency to the user through the use of icons, colors, and descriptive indicators (e.g. Soon, Kind of soon, Not soon, Inactive)

## Related Issue



Closes #13 

## Acceptance Criteria


- Items in the list are shown with an indicator that tells the user they should buy the item “soon”, “kind of soon”, or “not soon”; or that the item is “inactive”
   - This urgency indicator does not rely on only color
 - api/firestore.js exports a new comparePurchaseUrgency function with the following behaviors
   - sorts inactive items last, then
   - sorts items in ascending order of days until purchase, and
   - sorts items with the same days until purchase alphabetically

## Type of Changes

Enhancement

## Updates

### Before


<img width="728" alt="Screenshot 2024-03-15 at 23 37 22" src="https://github.com/the-collab-lab/tcl-72-smart-shopping-list/assets/103657276/9f761dc5-9a25-4711-9fa2-7d96f88a7050">

### After


<img width="690" alt="Screenshot 2024-03-15 at 23 16 09" src="https://github.com/the-collab-lab/tcl-72-smart-shopping-list/assets/103657276/eb638497-6359-4280-9833-bca641ddd019">

## Testing Steps / QA Criteria


1. Clone the branch ij-kn-sort-items
2. Go to the list page
3. Manipulate data on Firebase to see the urgency indicator next to each item:
  - Soon = 7 days or fewer until the next purchase
  - Kind of soon = between 7 & 30 days until the next purchase
  - Not soon = 30 days or more until the next purchase
  - Inactive = 60 days have passed since the last purchase
4. To see if items are sorted alphabetically when they have the same purchase urgency:
  - Add 2 items
  - Manipulate data in Firebase to have both items have the same daysTillNextPurchase
  - You should see items sorted alphabetically
